ansible/roles/flannel/tasks/client.yml

```diff
@@ -4,15 +4,33 @@
   args:
         name: flannel
         state: latest
-  when: not is_atomic and not is_coreos
+  when: not is_atomic and not is_coreos and ansible_distribution != "Ubuntu"
+
+- name: Force flannel_source_type to github when packages are not available
+  set_fact:
+    flannel_source_type: "github-release"
```

Comment:
Default "flannel_source_type" is defined in roles/flannel/defaults/main.yaml, so here it just overrides to "github-release" for coreos and ubuntu.